File: 0404-sum-of-left-leaves/0404-sum-of-left-leaves.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution(object):
    def sumOfLeftLeaves(self, root):
        """
        :type root: TreeNode
        :rtype: int
        """
        
        summer = 0
        
        def searcher(noder, status):
            result = 0
            if noder.left:
                result += searcher(noder.left, "left")
            if noder.right:
                result += searcher(noder.right, "right")
            if noder.left == None and noder.right == None and status == "left":
                result += noder.val
            return result
        return(searcher(root,"right"))
                
        

File: 0404-sum-of-left-leaves/0404-sum-of-left-leaves.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    int answer = 0; 
    public int sumOfLeftLeaves(TreeNode root) {
        travel(root, false);
        return answer;
    }
    
    public void travel(TreeNode noder, Boolean lefter){
        if(noder.left == null && noder.right == null && lefter){
            answer += noder.val;
        }
        else{
            if(noder.left != null){
                travel(noder.left, true);
            }
            if(noder.right != null){
                travel(noder.right, false);
            }
        }
    }
}